src/platform/window.cpp

```diff
@@ -59,13 +59,17 @@ Window::Window(ContextType type) {
 
 	_window = glfwCreateWindow(1920, 1080, "", nullptr, nullptr);
 
+	glfwSetCursorPos(_window, 960, 540);
+	_lastMousePosition = glm::vec2(960, 540);
```

Comment:
Just an Idea here, but you could use the callback if the cursor is inside or outsidethe window and if it leaves the window (or is outside of it in the beginning), you set the last mouse position to an invalid state. If it enters the window again you take the first position and use it in the mouse cursor callback.